packages/cli/src/lib/bundler/packageDetection.ts

```diff
@@ -45,9 +45,19 @@ function readPackageDetectionConfig(
     return {};
   }
 
+  if (typeof packages !== 'object' || Array.isArray(packages)) {
+    throw new Error(
+      "Invalid config at 'app.experimental.packages', expected object",
+    );
+  }
+  const packagesConfig = new ConfigReader(
```

Comment:
You really didn't need to reconstruct this right? If you kept the original paths below it'd worked out anyway even without the `object` check etc above since `getOptionalStringArray` would have checked that the subpaths are objects anyway.

Reply:
Nop :<

It blows up here if you try to create an sub config: https://github.com/backstage/backstage/blob/bdfe792890ff87be023991a7dbc25865e7d7ea42/packages/config/src/reader.ts#L192

And here if trying to read string arrays with the full path: https://github.com/backstage/backstage/blob/bdfe792890ff87be023991a7dbc25865e7d7ea42/packages/config/src/reader.ts#L422

To be clear the case where it throws is if you have a fallback config with `packages: 'all'` and you're missing either `include` or `exclude` here